pd/src/pending_block.rs

```diff
@@ -16,14 +17,21 @@ pub struct PendingBlock {
     pub note_commitment_tree: NoteCommitmentTree,
     /// Stores note commitments for convienience when updating the NCT.
     pub notes: BTreeMap<note::Commitment, PositionedNoteData>,
+    /// Notes that are undelegation outputs in this block (not to be committed to the NCT yet).
+    pub undelegation_notes: BTreeMap<note::Commitment, QuarantinedNoteData>,
     /// Nullifiers that were spent in this block.
     pub spent_nullifiers: BTreeSet<Nullifier>,
+    /// Nullifiers that were associated with undelegation transactions in this block, associated
+    /// with their unbonding epoch and validator identity key.
+    pub quarantined_nullifiers: BTreeMap<Nullifier, u64>,
```

Comment:
What's the `u64` here, is that the unbonding epoch? If so, how are the nullifiers associated with the validator identity key?

Reply:
Replaced with `QuarantineGroup`, which stores all the important information.